tests/php/BlockTypes/ProductCollection.php

```diff
@@ -70,9 +70,7 @@ private function initialize_merged_query( $parsed_block = array() ) {
 
 		$query = build_query_vars_from_query_block( $block, 1 );
```

Comment:
I think we still need to review how we're setting up the test, as the `ProductCollection::initialize()` is still executed, resulting we're running the `build_frontend_query` method two times here. This is clearly different from real usage.  The same thing happens to `ProductQuery` tests.

The issue, however, doesn't fall into the scope of this PR. So if we agree to fix it, we can work on it in another PR.

Reply:
I completely agree with you. However, I don't think this is a high-priority task, so perhaps we can create an issue and work on it later. What do you think?

I will go ahead and merge this PR.

Reply:
I opened #10484 for this.